task/sast-shell-check/0.1/sast-shell-check.yaml

```diff
@@ -0,0 +1,187 @@
+      script: |
+        #!/usr/bin/env bash
+        set -x
+        . /utils.sh
+
+        # FIXME: once https://github.com/konflux-ci/konflux-test/pull/303 is available in latest image,
+        # work around below is no longer needed
+        microdnf install -y ShellCheck
+        microdnf download csmock-plugin-shellcheck && rpm -ivh --nodeps ./csmock-plugin-shellcheck-*.rpm && rm -f ./csmock-plugin-shellcheck-*.rpm
+
+        PACKAGE_VERSION=$(rpm -q --queryformat '%{NAME}-%{VERSION}-%{RELEASE}\n' ShellCheck)
+
+        OUTPUT_FILE="shellcheck-results.json"
+        SOURCE_CODE_DIR=$(workspaces.workspace.path)/source
+
+        # generate all shellcheck result JSON files to $SC_RESULTS_DIR, which defaults to ./shellcheck-results/
+        /usr/share/csmock/scripts/run-shellcheck.sh "$SOURCE_CODE_DIR"
+
+        CSGREP_OPTS=(
+            --mode=json
+            --strip-path-prefix="$SOURCE_CODE_DIR"/
+            --remove-duplicates
+            --embed-context=3
+            --set-scan-prop="ShellCheck:${PACKAGE_VERSION}"
+        )
+        if [[ "$IMP_FINDINGS_ONLY" == "true" ]]; then
+            # predefined list of shellcheck important findings
+            CSGREP_EVENT_FILTER='\[SC(1020|1035|1054|1066|1068|1073|1080|1083|1099|1113|1115|1127|1128|1143|2043|2050|'
+            CSGREP_EVENT_FILTER+='2055|2057|2066|2069|2071|2077|2078|2091|2092|2157|2171|2193|2194|2195|2215|2216|'
+            CSGREP_EVENT_FILTER+='2218|2224|2225|2242|2256|2258|2261)\]$'
+            CSGREP_OPTS+=(
+                --event="$CSGREP_EVENT_FILTER"
+            )
+        else
+            CSGREP_OPTS+=(
+                --event="info|warning"
+            )
+        fi
+
+        csgrep "${CSGREP_OPTS[@]}" ./shellcheck-results/*.json > "$OUTPUT_FILE"
+
+        if [ $? -ne 0 ]; then
+            echo "Error occurred while running 'run-shellcheck.sh'"
+            note="Task $(context.task.name) failed: For details, check Tekton task log."
+            ERROR_OUTPUT=$(make_result_json -r ERROR -t "$note")
+            echo "${ERROR_OUTPUT}" | tee "$(results.TEST_OUTPUT.path)"
+            exit 1
+        fi
+
+        # Filter known false positives if KFP_GIT_URL is set
+        if [ -n "${KFP_GIT_URL}" ]; then
+            echo "Filtering known false positives using ${KFP_GIT_URL}"
+
+            # build initial csfilter-kfp command
+            csfilter_cmd="csfilter-kfp --verbose --kfp-git-url=${KFP_GIT_URL}"
+
+            # Append --project-nvr option if PROJECT_NVR is set
+            if [ -n "${PROJECT_NVR}" ]; then
+                csfilter_cmd+=" --project-nvr=${PROJECT_NVR}"
+            fi
+
+            if [[ "${RECORD_EXCLUDED}" == "true" ]]; then
+                csfilter_cmd+=" --record-excluded=excluded-findings.json"
+            fi
+
+            # Execute the command and capture any errors
+            if ! "${csfilter_cmd}" "${OUTPUT_FILE}" > "${OUTPUT_FILE}.filtered" 2> "${OUTPUT_FILE}.error"; then
```

Comment:
I do not think this could work.  `"${csfilter_cmd}"` will be treated by bash as a name of a single command rather than a command followed by a list of arguments.

Reply:
indeed, array is a better option here than string. Updated in https://github.com/konflux-ci/build-definitions/pull/1437/commits/5bbfa2a1ca5dcbd83d6d94ca8866e54070c7a0ce